core/src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -55,7 +56,8 @@
  *
  * @author Daniel Kayiwa, Drew Roos
  */
-public class FormDef implements IFormElement, Localizable, Persistable, IMetaData {
+public class FormDef extends ActionController
```

Comment:
@ctsims The most recent commit makes the suggested change. I'm not sure whether or not I find it better. There are some trade-offs in terms of now having to repeat the same code in both `FormDef` and `QuestionDef`, but I do agree that having the original way was a bit heavy-handed for representing the true state of things. Up to you which seems better, or if the change I made can be improved upon still.